lib/line-numbers.js

```diff
@@ -0,0 +1,147 @@
+'use strict';
+
+const callsites = require('callsites');
+const sourceMapSupport = require('source-map-support');
+const parseTestSourceInFile = require('./test-ast');
+
+module.exports = {
+  parseLineNumbers,
+  parseLineNumbersInPath,
+  parseLineNumbersInPaths,
+  hasLineNumbersSuffix,
+  stripLineNumbersSuffix,
+  resolveEndLineNumberForTestInFile,
+  getLineNumberRangeForTestInFile,
+  isTestSelectedByLineNumbers
+};
+
+const NUMBER_REGEX = /^-?\d+$/;
+const RANGE_REGEX = /^(?<start>-?\d+)-(?<end>-?\d+)$/;
+const LINE_NUMBERS_REGEX = /^(-?\d+(-\d+)?,?)+$/
+const PATH_DELIMITER = ':';
+
+const parseNumber = string => parseInt(string, 10);
+const range = (start, end) => Array(end - start + 1).fill(start).map((e, i) => e + i);
+const distinctArray = array => [...new Set(array)];
+const removeAllWhitespace = string => string.replace(/\s/g, "");
+const pluralizeIf = (string, condition) => condition ? `${string}s` : string;
+const getLineNumbersSuffix = path => path.split(PATH_DELIMITER).pop();
```

Comment:
Perhaps this was discussed and we didn't consider the edge case worthwhile, but do we need to handle colons in the filename? Perhaps before the file extension?

Reply:
If I understand you correctly, that's already handled. See tests [`file delimiter without numbers in file name`](https://github.com/avajs/ava/pull/2300/files#diff-d4bc66da908e50f18b3a3a9cc95039ebR169) and [`file delimiter and numbers in file name`](https://github.com/avajs/ava/pull/2300/files#diff-d4bc66da908e50f18b3a3a9cc95039ebR174).

If I've missed the point, would you mind demonstrating with an example?

Reply:
@novemberborn Please resolve if you feel this is taken care of. Otherwise, would you mind providing more details/examples?